src/modules/AddOnProduct.rb

```diff
@@ -2235,7 +2235,7 @@ def add_product_from_cd(url, pth, priority, prodname, check_name)
         end
 
         # ask for a different medium
-        current_url = AskForCD(current_url, prodname)
+        current_url = AskForCD(url, prodname)
```

Comment:
That's the point: in the first iteration, `current_url` is set to `cd:/?devices=/dev/sr0` (and `url` and `current_url` won't point to the same string anymore).

In the second iteration, `AskForCD` will receive that value and it will append `devices=/dev/sr0`  ( because of [L1553](https://github.com/imobachgs/yast-packager/blob/5e7210c0614eb08eb51a6e105be4280c005cc92e/src/modules/AddOnProduct.rb#L1553) ) resulting in `cd:/?devices=/dev/sr0&devices=/dev/sr0`. And so on the third, fourth... iterations.